cgp/genome.py

```diff
@@ -270,6 +269,105 @@ def randomize(self, rng: np.random.RandomState) -> None:
         # accept generated dna if it is valid
         self.dna = dna
 
+    def splice_dna(self, dna_insert: List[int], internal_start_node: int = 0) -> int:
+        """Splice the dna.
+
+        Replaces a segment of the dna starting from internal start node with the dna insert
+
+        Parameters
+        ----------
+        dna_insert: List[int]
+            dna segment to be inserted.
+        internal_start_node: int
+            idx of the internal node, where the insert starts.
+            Defaults to 0.
+
+        Returns
+        ----------
+        address_last_inserted_node: int
+            address of the last node inserted.
+        """
+        if len(dna_insert) % self._length_per_region != 0 or len(dna_insert) == 0:
+            raise ValueError("Length of inserted dna must be a multiple of region length")
```

Comment:
```suggestion
            raise ValueError(f"Length of inserted dna must be a multiple of region length ({self._length_per_region})")
```

provides a bit more info to the user